irony-cdb.el

```diff
@@ -114,6 +128,24 @@ for files that it cannot handle."
 ;; Functions
 ;;
 
+(defun irony-cdb--locate-dominating-file-with-dirs (file name subdirectories)
+  "Convenience wrapper around `locate-dominating-file'
+
+Looks up the directory hierarchy from FILE for to locate any directory
+in `subdirectories` which contains NAME. Stops at the first parent
+directory where the test succeeds and returns the full path to file
+found."
+  (let ((check-subdirs
+         (lambda (dir)
+           (when dir
+             (cl-loop for subdir in subdirectories
+                      for match =(concat dir "/" subdir "/" name)
```

Comment:
formatting: I would add a space between `=` and `(`.
use `(file-name-as-directory FILE)` instead of hardcoding slashes.